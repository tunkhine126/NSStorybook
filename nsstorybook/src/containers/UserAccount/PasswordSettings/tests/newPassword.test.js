/**
 *
 * Tests for NewPassword
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Form } from 'informed';

import TestingComponent from 'components/shared/Testing';
import NewPassword from '../newPassword';

Enzyme.configure({ adapter: new Adapter() });

describe('<NewPassword />', () => {
  it('Simulate Password change', () => {
    const spy = jest.spyOn(global.console, 'error');

    const newValue = 'test';
    const wrapper = mount(
      <TestingComponent>
        <Form id="test-form">
          <NewPassword />
        </Form>
      </TestingComponent>
    );
    const password = wrapper.find('input').first();
    const passwordConfirm = wrapper.find('input').at(1);

    password.simulate('change', { target: { value: newValue } });
    password.simulate('change', { target: { value: 'Testing12345' } });
    passwordConfirm.simulate('change', { target: { value: 'Testing12345' } });

    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NewPassword />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <NewPassword />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
