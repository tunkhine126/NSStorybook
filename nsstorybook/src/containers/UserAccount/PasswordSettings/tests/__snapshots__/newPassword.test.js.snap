// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewPassword /> Should render and match the snapshot 1`] = `
<div
  class="MuiGrid-root makeStyles-spacing-8 MuiGrid-item"
>
  <p
    class="MuiTypography-root makeStyles-darkText-1 MuiTypography-body1 MuiTypography-gutterBottom"
  >
    <span>
      Please make sure your password is strong and 6 characters minimum.
    </span>
  </p>
  <div
    class="styles__ComponentStyle-zscvjs-0 iXrycp"
  >
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-input-260 makeStyles-root-264 MuiFormControl-marginNormal"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink="false"
        for="password"
        id="password-label"
      >
        <span>
          New Password
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root makeStyles-root-153 MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input"
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="styles__ComponentStyle-zscvjs-0 iXrycp"
  >
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-input-265 makeStyles-root-269 MuiFormControl-marginNormal"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
        data-shrink="false"
        for="passwordConfirmation"
        id="passwordConfirmation-label"
      >
        <span>
          Confirm Your New Password
        </span>
      </label>
      <div
        class="MuiInputBase-root MuiFilledInput-root makeStyles-root-153 MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiFilledInput-input"
          id="passwordConfirmation"
          name="passwordConfirmation"
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;
